config/configauth/clientauth.go

```diff
@@ -15,22 +15,8 @@
 package configauth // import "go.opentelemetry.io/collector/config/configauth"
 
 import (
-	"net/http"
-
-	"google.golang.org/grpc/credentials"
-
-	"go.opentelemetry.io/collector/component"
+	"go.opentelemetry.io/collector/extension/auth"
 )
 
-// ClientAuthenticator is an Extension that can be used as an authenticator for the configauth.Authentication option.
-// Authenticators are then included as part of OpenTelemetry Collector builds and can be referenced by their
-// names from the Authentication configuration.
-type ClientAuthenticator interface {
-	component.Extension
-
-	// RoundTripper returns a RoundTripper that can be used to authenticate HTTP requests.
-	RoundTripper(base http.RoundTripper) (http.RoundTripper, error)
-
-	// PerRPCCredentials returns a PerRPCCredentials that can be used to authenticate gRPC requests.
-	PerRPCCredentials() (credentials.PerRPCCredentials, error)
-}
+// Deprecated: [v0.67.0] Use auth.Client
+type ClientAuthenticator = auth.Client
```

Comment:
Move all these "deprecated" types in a file "deprecated.go". Then git will show the other files as moved, and history will be preserved.